### ISSUE TITLE
bug: [1.7] Button Click events seems to produce a table instead of a string

### ISSUE THREAD

**Issue by ajh123** [Pyroxenium/Basalt#55]
**Describe the bug**

When a button gets clicked, instead of a string like `"mouse_click"` or `"mouse_up"` the `event` (2nd parameter) is a table.

**To Reproduce**

1. A small program can be used to test the behaviour:
```lua
local basalt = require("basalt")

local main = basalt.createFrame()

main:addButton()
:setText("Start")
:setPosition(1, 1)
:setSize("7", "1")
:onClick(function(self, event, button, x, y)
	basalt.debug("Event ("..tostring(event)..") = {")
	if type(event) == "table" then
		for k, v in pairs(event) do
			basalt.debug(k.."="..tostring(v)..",")
		end
	end
	basalt.debug("}")
end)

basalt.autoUpdate()
```
2. Run the program.
2. Click the "Start" button.
4. Open the Basalt debug window.
5. See something like the attached screenshot. 

**Expected behavior**

Once the button is clicked, the Basalt debug window will contain:
```
Event (mouse_click) = {
}
``` 

**Screenshots**

![basalt_error2](https://user-images.githubusercontent.com/41990982/235739094-26c5ba0d-3290-43c7-a4fa-147adcaa1a50.PNG)

^ The contents of the `event`.

**Additional context**

N/A ?

**Checklist**

[ X ] I am running the latest version.
Tick the box if you are running the latest version!

**Comment by NoryiE** [comment]
Hello, thank you i will look into this asap!

**Comment by NoryiE** [comment]
The issue should be fixed now! Here's the fix: [a97254d](https://github.com/Pyroxenium/Basalt/commit/a97254ddf781a96fadd7cf7f26efc5e8076fd8b3)

**Comment by ajh123** [comment]
Thanks, it works now.
